Smt/Reconstruct/Real.lean

```diff
@@ -523,6 +622,271 @@ where
       addThm q(($x * $y ≤ $b * $x + $a * $y - $a * $b) = ($x ≤ $a ∧ $y ≥ $b ∨ $x ≥ $a ∧ $y ≤ $b)) q(@Real.mul_tangent_lower_eq $a $b $x $y)
     else
       addThm q(($x * $y ≥ $b * $x + $a * $y - $a * $b) = ($x ≤ $a ∧ $y ≤ $b ∨ $x ≥ $a ∧ $y ≥ $b)) q(@Real.mul_tangent_upper_eq $a $b $x $y)
-  | _ => return none
+  | .ARITH_TRANS_PI =>
+    let l : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    let u : Q(Real) ← reconstructTerm pf.getArguments[1]!
+    addTac q(Real.pi ≥ $l ∧ Real.pi ≤ $u) TransFns.arithTransPiTac
+  | .ARITH_TRANS_EXP_ZERO =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q(($t = 0) = (Real.exp $t = 1)) q(TransFns.arithTransExpZeroEq $t)
+  | .ARITH_TRANS_SINE_SHIFT =>
+    let x : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    let s : Q(Real) := q(Classical.epsilon (TransFns.shift_prop_part $x))
+    let y : Q(Real) := q(Classical.epsilon (TransFns.shift_prop $x $s))
+    addThm q(TransFns.shift_prop $x $s $y) q(TransFns.arithTransSineShift₁ $x)
+  | .ARITH_TRANS_EXP_POSITIVITY =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q(Real.exp $t > 0) q(TransFns.arithTransExpPositivity $t)
+  | .ARITH_TRANS_SINE_TANGENT_PI =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q(($t > (-1) * Real.pi → Real.sin $t > (-1) * Real.pi - $t) ∧ ($t < Real.pi → Real.sin $t < Real.pi - $t)) q(TransFns.arithTransSineTangentPi $t)
+  | .ARITH_TRANS_SINE_TANGENT_ZERO =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q(($t > 0 → Real.sin $t < $t) ∧ ($t < 0 → Real.sin $t > $t)) q(TransFns.arithTransSinTangentZero $t)
+  | .ARITH_TRANS_EXP_SUPER_LIN =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q($t ≤ 0 ∨ Real.exp $t > $t + 1) q(TransFns.arithTransExpSuperLin $t)
+  | .ARITH_TRANS_EXP_NEG =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q(($t < 0) = (Real.exp $t < 1)) q(TransFns.arithTransExpNeg' $t)
+  | .ARITH_TRANS_SINE_BOUNDS =>
+    let t : Q(Real) ← reconstructTerm pf.getArguments[0]!
+    addThm q((Real.sin $t ≤ 1 ∧ Real.sin $t ≥ -1)) q(TransFns.arithTransSineBounds $t)
+  | .ARITH_TRANS_EXP_APPROX_BELOW =>
+    let d : Q(Int) ← reconstructTerm pf.getArguments[0]!
```

Comment:
I see that all use cases of `d` are inside `Int.natAbs` applications. In that case, it's better to just get its value as a `Nat` as follows:
```suggestion
    let d : Nat := pf.getArguments[0]!.getIntegerValue!.natAbs
```
The quotation library can automatically lift values of type `Nat` into the corresponding `Q(Nat)` expression as needed!